console/v1alpha1/types_console_plugin.go

```diff
@@ -37,31 +37,65 @@ type ConsolePluginSpec struct {
 	// proxy is a list of Services that the plugin needs to connect to.
 	// +kubebuilder:validation:Optional
 	// +optional
-	Proxy ConsolePluginProxy `json:"proxy"`
+	Proxy ConsolePluginProxy `json:"proxy,omitempty"`
 }
 
 // ConsolePluginProxy holds information on various service types
 // to which console's backend will proxy the plugin's requests.
 type ConsolePluginProxy struct {
-	// services is a list of in-cluster Services that the plugin
-	// will connect to. The Service must use HTTPS. Console backend
-	// exposes the following endpoint in order to proxy communication
-	// between the plugin and the Service:
+	// type is the type of the console plugin's proxy
+	// +kubebuilder:validation:Required
+	// +required
+	Type ConsolePluginProxyType `json:"type"`
+	// alias is an unique proxy name that identifies the plugin's proxy
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	// +required
+	Alias string `json:"alias"`
+	// service is a in-cluster Services that the plugin will connect to.
+	// The Service must use HTTPS. Console backend exposes the following
+	// endpoint in order to proxy communication between the plugin and the Service:
 	//
-	// /api/proxy/namespace/<service-namespace>/service/<service-name>:<port-number>/<request-path>?<optional-query-parameters>
+	// /api/proxy/namespace/<plugin-name>/<proxy-alias>/<request-path>?<optional-query-parameters>
 	//
 	// Request example path:
 	//
-	// /api/proxy/namespace/helm/service/helm-charts:8443/releases?limit=10
+	// /api/proxy/acm/search/pods?namespace=openshift-apiserver
 	//
 	// +kubebuilder:validation:Optional
 	// +optional
-	Services []ConsolePluginProxyService `json:"services"`
+	Service ConsolePluginProxyServiceConfig `json:"service,omitempty"`
```

Comment:
I think this is OK, but doesn't seem to match what was in https://github.com/openshift/enhancements/pull/967. We should update the enhancement.